lib/pipefy_message/publisher.rb

```diff
@@ -14,8 +14,12 @@ def initialize(broker = "aws", broker_opts = {})
       @publisher_instance = build_publisher_instance
     end
 
-    def publish(message, topic, context = nil)
-      @publisher_instance.publish(message, topic, context)
+    def publish(message, topic, context = nil, correlation_id = nil)
```

Comment:
Reposting a comment I'd left in a previous version, which GitHub has now hidden:

**question:** I've been thinking about this context and I feel it's possibly a little confusing, given that we already have a `context: async_processing` in our logs (this would be nested one level deeper than that - `{ ... context: async_processing, ... message: { ... context: ... } }` ). Should we rename this to `message_context`, to highlight it's related to where the event came from? Or perhaps even drop it altogether, now that we have correlation and event IDs (though this could potentially be more human-readable; `context: formula_fields, correlation_id: ...`)? WDYT?

Reply:
The idea of this field was to receive a context(something that can give more info about the event/operation) during the publish flow and more than just log, send it to the consumer, to help in tracing and stuff like that. But to be honest, it was never used, so... I agree that now that we have more specific topic and queues names and also those new attributes (correlation and event ID) this context field became useless and can be dropped.

Reply:
**suggestion:** Have you considered having `correlation_id` inside the `context` hash? The caller will need to provide this anyways... It wouldn't be much different appending this property instead of sending an extra argument.

Reply:
@guustavov I had considered that and discussed it with @jleber before, but, unfortunately, it seems that AWS SNS messages [don't support](https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html) JSON objects as attributes, only strings, numbers, binary data and whatever a `string.array` is (sth like a C "string"?), so I don't see what we could send a hash as. I mean, I suppose we could send it as a string, but that would require an additional parsing step to get it back to a structured format on the other end, so perhaps just having them as separate attributes is better.